lib/session/DatawrapperSession.php

```diff
@@ -62,6 +62,8 @@ public static function initSession() {
             $cookieOpts['samesite'] = (isset($_SESSION['type']) && $_SESSION['type']
                 == 'token') ? 'None' : $sameSite;
             setcookie($ses, $_COOKIE[$ses], $cookieOpts);
+        } else {
+            $_COOKIE[$ses] = session_id();
```

Comment:
I've read up a little bit by now and assuming that `session_id()` will always hold a valid ID (and i guess that's what line 56 ensures), this should be working.

Hmm.

If a session is present when we hit line 59 we do things to the cookie (expires, samesite…).
Now, if the session was not present, we set it - but we don't do the above transformations to the cookie in that case.

Is this what we want?

Reply:
Yes. Keep in mind that transformation doesn't alter the `$_COOKIE` object in any way, it only uses that to determine the presence of a request cookie, and in turn alters the _response cookie_  via `setcookie`. We don't need to alter the response cookie in any way here.